static/src/javascripts/projects/common/modules/commercial/ads/user-ad-feedback.js

```diff
@@ -0,0 +1,48 @@
+define([
+    'bonzo',
+    'qwery',
+    'common/utils/fetch',
+    'common/utils/config'
+], function (
+    bonzo,
+    qwery,
+    fetch,
+    config
+) {
+
+    function userAdFeedback(pagePath, adSlotId, creativeid, reason) {
+        // TODO: This. Better.
+        if (reason !== 'ad-feedback-menu-opened') {
+            return recordUserAdFeedback(pagePath, adSlotId, creativeid, reason);
+        } else {
+            return recordUserAdFeedback(pagePath, adSlotId, creativeid, 'ad-feedback-menu-opened');
+        }
+    }
+
+    function recordUserAdFeedback(pagePath, adSlotId, creativeId, feedbackType) {
+        var feedbackUrl = 'https://j2cy9stt59.execute-api.eu-west-1.amazonaws.com/prod/adFeedback';
+        var stage = config.page.isProd ? 'PROD' : 'CODE';
+        var data = '{"stage":"'+stage+'", "adPage":"'+pagePath+'", "adSlotId":"'+adSlotId+'", "adCreativeId":"'+creativeId+'", "adFeedback":"'+feedbackType+'"}';
+        return fetch(feedbackUrl, {
+            method: 'post',
+            body: data,
+            mode: 'cors'
+        }).then(function() {    // ignore the response
```

Comment:
`Promise.then` accepts two callbacks: one when the promise is kept, one when the promise is broken (`.catch` is just sugar). You can avoid superfluous ticks this way:

```
fetch(...).then(onComplete, onComplete);
function onComplete() {
   if( feedbackType ...
}
```